### ISSUE TITLE
Date filter in dashboards doesn't work as expected

### ISSUE THREAD

**Issue by cyberhorse** [metabase/metabase#5781]
Date parameter in dashboard cannot be linked against a date variable in a question.

Simple way to reproduce the issue:
- Create a basic question on a table that has a date field
- Add that question to a dashboard and add a date filter, linked to the date field above (so far all good)
- Edit the original question and convert to SQL query
  - Date filter in the dashboard no longer works (says "No valid fields")
- Add a date variable to the question/SQL query (hoping it will expose a date filter in the dashboard)
  - The variable isn't recognised in the dashboard as something that can be linked against the date parameter in the dashboard

The two expected behaviour above maybe two separate issues.

- Database: mariadb
- Metabase version: 25.2
- Browser: Chrome

**Comment by cyberhorse** [comment]
After some head bashing, I found out that I change the date variable to field filter variable, it let's me link it to a date parameter in the dashboard. I believe if it's a date variable, it should also work (or at least have some guidance in the dashboard to clarify the variable type should be changed)

**Comment by mazameli** [comment]
Good feedback. This flow could definitely be clearer. 

The reason you can't do this with a "normal" date variable is because it wouldn't work for all types of dashboard date filters: specifically, it wouldn't work for date filters that require two values, like a Between filter (starting date and ending date). Field filters accept a whole clause, while date variables only accept a single value.

**Comment by cyberhorse** [comment]
Yeah, I guessed as much after I got it to work. It's just really frustrating when I thought I had done everything right (I have a date variable, I want to link it to a date filter), but it kept saying no valid fields ... :)

**Comment by billyY2** [comment]
How do you convert the date variable to a field filter variable?

**Comment by billyY2** [comment]
It looks like you can only do a field filter variable on a table that is already populated. In other words it can't be a field filter on one of the fields you have created in the metabase query. Is that correct?

**Comment by asyncopation** [comment]
@billyY2 Yes, I believe you can only associate the Field Filter with an existing field in the database schema.

---

> The reason you can't do this with a "normal" date variable is because it wouldn't work for all types of dashboard date filters: specifically, it wouldn't work for date filters that require two values, like a Between filter (starting date and ending date). Field filters accept a whole clause, while date variables only accept a single value.

@mazameli I wonder if it would be possible to standardize the variable names to accommodate a date range for an SQL question using the Date filter type?  For instance {{ date_start }} and {{ date_end }}.

A common use case we're running into is one where our "Start" and "End" date filters are applied to multiple columns to pull in joined data sets over the same time frame.  For instance:

```sql
SELECT fields...
FROM table_a
JOIN table_b...
WHERE table_a.created_at BETWEEN {{ date_start }} and {{ date_end }}
AND table_b.created_at BETWEEN {{ date_start }} and {{ date_end }}
```

This way the user can change a start and end date filter once, and they are applied to multiple fields.  We would then like to use these as part of a dashboard filter.

However, since dashboard filters currently only accept "Field Filters", and these are tied to a single field/column, this will not work.

Creating 2 Field Filters (one for `table_a.created_at` and `table_b.created_at`) would require date entry twice on the question itself (a minor awkward inconvenience), but still would not be effective since dashboard filters will only accept a single Field Filter association.

Thanks for any insight!

**Comment by flamber** [comment]
Related to #4509/#3156 and #3324
